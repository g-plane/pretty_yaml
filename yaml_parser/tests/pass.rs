use insta::{assert_snapshot, glob, Settings};
use std::{fs, path::Path};

#[test]
fn pass_snapshot() {
    glob!("pass/*.yaml", |path| {
        let input = fs::read_to_string(path).unwrap();
        let tree = match yaml_parser::parse(&input) {
            Ok(tree) => tree,
            Err(err) => panic!("failed to parse '{}': {err}", path.display()),
        };
        assert_eq!(
            tree.to_string(),
            input,
            "syntax tree of '{}' does not match source",
            path.display()
        );

        build_settings(path).bind(|| {
            let name = path.file_stem().unwrap().to_str().unwrap();
            assert_snapshot!(name, format!("{tree:#?}"));
        });
    });
}

fn build_settings(path: &Path) -> Settings {
    let mut settings = Settings::clone_current();
    settings.set_snapshot_path(path.parent().unwrap());
    settings.remove_snapshot_suffix();
    settings.set_prepend_module_to_snapshot(false);
    settings.remove_input_file();
    settings.set_omit_expression(true);
    settings.remove_input_file();
    settings.remove_info();
    settings
}
