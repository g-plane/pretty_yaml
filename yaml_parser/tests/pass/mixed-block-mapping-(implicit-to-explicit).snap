---
source: yaml_parser/tests/pass.rs
---
ROOT@0..16
  DOCUMENT@0..15
    BLOCK@0..15
      BLOCK_MAP@0..15
        BLOCK_MAP_ENTRY@0..6
          BLOCK_MAP_KEY@0..1
            FLOW@0..1
              PLAIN_SCALAR@0..1 "a"
          COLON@1..2 ":"
          WHITESPACE@2..3 " "
          BLOCK_MAP_VALUE@3..6
            FLOW@3..6
              PLAIN_SCALAR@3..6 "4.2"
        WHITESPACE@6..7 "\n"
        BLOCK_MAP_ENTRY@7..15
          BLOCK_MAP_KEY@7..10
            QUESTION_MARK@7..8 "?"
            WHITESPACE@8..9 " "
            FLOW@9..10
              PLAIN_SCALAR@9..10 "d"
          WHITESPACE@10..11 "\n"
          COLON@11..12 ":"
          WHITESPACE@12..13 " "
          BLOCK_MAP_VALUE@13..15
            FLOW@13..15
              PLAIN_SCALAR@13..15 "23"
  WHITESPACE@15..16 "\n"
